avatars/client.py

```diff
@@ -123,4 +122,28 @@ def request(
         if result.headers["content-type"] == "application/json":
             return result.json()
         else:
-            return result.content
+            return result.text
+
+    def _get_file_argument(
+        self, file: Optional[Union[StringIO, BytesIO]]
+    ) -> Optional[Dict[str, Tuple[str, bytes, str]]]:
+
+        if not file:
+            return
+
+        filename = str(Path(file.name).name) if hasattr(file, "name") else "file.csv"
+
+        content = file.read(MAX_FILE_LENGTH)
+
+        if sys.getsizeof(content) >= MAX_FILE_LENGTH:
+            raise FileTooLarge(
+                f"The file size must not exceed {MAX_FILE_LENGTH / 1024 :.0f} MB."
+            )
+
+        encoded_content = (
+            content if isinstance(content, bytes) else content.encode("utf-8")
+        )
+
+        # The dictionary key must be 'file' and you MUST pass in a filename
+        # else there is a request ValidationError on the server side
+        return {"file": (filename, encoded_content, "text/csv")}
```

Comment:
Otherwise we could give it the file descriptor and let it handle encoding?

https://www.python-httpx.org/quickstart/#sending-multipart-file-uploads